src/core/tests/type_prop/is_inf.cpp

```diff
@@ -0,0 +1,91 @@
+        << "The output shape of IsInf is incorrect, it should be the same as shape of input data";
+}
+
+TEST(type_prop, is_inf_scalar) {
+    const auto data = make_shared<Parameter>(element::f32, PartialShape{1});
+    const auto is_inf = make_shared<IsInf>(data, IsInf::Attributes{});
+
+    EXPECT_EQ(is_inf->get_element_type(), element::boolean)
+        << "The output element type of IsInf should always be boolean";
+    EXPECT_EQ(is_inf->get_output_partial_shape(0), PartialShape({1}))
+        << "The output shape of IsInf is incorrect, it should be the same as shape of input data";
+}
+
+TEST(type_prop, is_inf_bfloat16) {
+    const auto data = make_shared<Parameter>(element::bf16, PartialShape{1, 64, 256, 256});
+    const auto is_inf = make_shared<IsInf>(data, IsInf::Attributes{});
+
+    EXPECT_EQ(is_inf->get_element_type(), element::boolean)
+        << "The output element type of IsInf should always be boolean";
+    EXPECT_EQ(is_inf->get_output_partial_shape(0), PartialShape({1, 64, 256, 256}))
+        << "The output shape of IsInf is incorrect, it should be the same as shape of input data";
+}
+
+TEST(type_prop, is_inf_float16) {
+    const auto data = make_shared<Parameter>(element::f16, PartialShape{1, 64, 256, 256});
+    const auto is_inf = make_shared<IsInf>(data, IsInf::Attributes{});
+
+    EXPECT_EQ(is_inf->get_element_type(), element::boolean)
+        << "The output element type of IsInf should always be boolean";
+    EXPECT_EQ(is_inf->get_output_partial_shape(0), PartialShape({1, 64, 256, 256}))
+        << "The output shape of IsInf is incorrect, it should be the same as shape of input data";
+}
+
+TEST(type_prop, is_inf_float64) {
+    const auto data = make_shared<Parameter>(element::f64, PartialShape{1, 64, 256, 256});
+    const auto is_inf = make_shared<IsInf>(data, IsInf::Attributes{});
+
+    EXPECT_EQ(is_inf->get_element_type(), element::boolean)
+        << "The output element type of IsInf should always be boolean";
+    EXPECT_EQ(is_inf->get_output_partial_shape(0), PartialShape({1, 64, 256, 256}))
+        << "The output shape of IsInf is incorrect, it should be the same as shape of input data";
+}
+
+TEST(type_prop, is_inf_interval) {
+    const auto data = make_shared<Parameter>(element::f32, PartialShape{Dimension(2, 4), Dimension(-1, 5)});
+    const auto is_inf = make_shared<IsInf>(data, IsInf::Attributes{});
+
+    EXPECT_EQ(is_inf->get_element_type(), element::boolean)
+        << "The output element type of IsInf should always be boolean";
+    EXPECT_EQ(is_inf->get_output_partial_shape(0), PartialShape({Dimension(2, 4), Dimension(-1, 5)}))
+        << "The output shape of IsInf is incorrect, it should be the same as shape of input data";
+}
+
+TEST(type_prop, is_inf_dynamic) {
+    const auto data = make_shared<Parameter>(element::f32, PartialShape::dynamic());
+    const auto is_inf = make_shared<IsInf>(data, IsInf::Attributes{});
```

Comment:
please also tests with non-default attributes `detect_negative` and `detect_positive`

Reply:
Added 4 additional cases to test all non-default attribute combinations